### PR TITLE
[EG] dont hardcode api_version on request

### PR DESCRIPTION
**Pull request by l0lawrence**

### CODE REVIEW

File: sdk/eventgrid/azure-eventgrid/azure/eventgrid/_publisher_client.py [hunk 1]
```diff
@@ -49,8 +49,6 @@
 if TYPE_CHECKING:
     # pylint: disable=unused-import,ungrouped-imports
     from azure.core.credentials import (
-        AzureKeyCredential,
-        AzureSasCredential,
         TokenCredential,
```
Comment (kashifkhan) [inline sdk/eventgrid/azure-eventgrid/azure/eventgrid/_publisher_client.py:52]: do you need to do the same with `TokenCredential`?

Comment (l0lawrence) [reply sdk/eventgrid/azure-eventgrid/azure/eventgrid/_publisher_client.py:52]: oh no I need to revert this actually sorry - they are all only used for typing purposes :)

File: sdk/eventgrid/azure-eventgrid/azure/eventgrid/_publisher_client.py [hunk 2]
```diff
@@ -79,6 +79,9 @@ class EventGridPublisherClient(object): # pylint: disable=client-accepts-api-ver
      implements SAS key authentication or SAS token authentication or a TokenCredential.
     :type credential: ~azure.core.credentials.AzureKeyCredential or ~azure.core.credentials.AzureSasCredential or
      ~azure.core.credentials.TokenCredential
+    :keyword api_version: Api Version. Default value is "2018-01-01". Note that overriding this
```
Comment (l0lawrence) [inline sdk/eventgrid/azure-eventgrid/azure/eventgrid/_publisher_client.py:82]:
```suggestion
    :keyword api_version: Api Version. Default value is the most recent api version. Note that overriding this
```

File: sdk/eventgrid/azure-eventgrid/azure/eventgrid/aio/_publisher_client_async.py [hunk 3]
```diff
@@ -73,6 +73,9 @@ class EventGridPublisherClient: # pylint: disable=client-accepts-api-version-key
      SAS key authentication or SAS token authentication or an AsyncTokenCredential.
     :type credential: ~azure.core.credentials.AzureKeyCredential or ~azure.core.credentials.AzureSasCredential or
      ~azure.core.credentials_async.AsyncTokenCredential
+    :keyword api_version: Api Version. Default value is "2018-01-01". Note that overriding this
```
Comment (l0lawrence) [inline sdk/eventgrid/azure-eventgrid/azure/eventgrid/aio/_publisher_client_async.py:76]:
```suggestion
    :keyword api_version: Api Version. Default value is the most recent api version. Note that overriding this
```

File: sdk/eventgrid/azure-eventgrid/azure/eventgrid/aio/_publisher_client_async.py [hunk 4]
```diff
@@ -73,6 +74,9 @@ class EventGridPublisherClient: # pylint: disable=client-accepts-api-version-key
      SAS key authentication or SAS token authentication or an AsyncTokenCredential.
     :type credential: ~azure.core.credentials.AzureKeyCredential or ~azure.core.credentials.AzureSasCredential or
      ~azure.core.credentials_async.AsyncTokenCredential
+    :keyword api_version: Api Version. Default value is the most recent api version. Note that overriding this
```
Comment (l0lawrence) [inline sdk/eventgrid/azure-eventgrid/azure/eventgrid/aio/_publisher_client_async.py:77]:
```suggestion
    :keyword api_version: Api Version.  Will default to the most recent Api Version. Note that overriding this
```

**Pull request merged**